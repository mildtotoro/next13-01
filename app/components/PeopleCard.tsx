import Link from 'next/link'
import React from 'react'

const PeopleCard = ({ data }: Record<string, any>) => {
  return (
    <div
      key={data.name}
      className="border border-solid border-gray-300 w-[250px] m-3 p-4"
    >
      <h2 data-testid="name">{data.name}</h2>
      <h3>Birth Year: {data.birth_year}</h3>
      <Link href={data.homeworld} target="_blank">
        Home world
      </Link>
      <div></div>
      <Link href={data.url} target="_blank">
        Read more
      </Link>
    </div>
  )
}

export default PeopleCard
