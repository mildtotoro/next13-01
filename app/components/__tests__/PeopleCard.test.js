import { render, screen } from '@testing-library/react'
import PeopleCard from '../PeopleCard'
import '@testing-library/jest-dom'

describe('PeopleCard', () => {
  it('renders a card', () => {
    render(
      <PeopleCard
        data={{
          id: 'b66231c5-3f2c-4997-91ca-49aa42b8781f',
          name: 'jendi asa',
          height: '150',
          mass: '49',
          hair_color: 'brown',
          skin_color: 'light',
          eye_color: 'brown',
          birth_year: '19BBY',
          gender: 'female',
          homeworld: 'https://swapi.dev/api/planets/2/',
          films: [
            'https://swapi.dev/api/films/1/',
            'https://swapi.dev/api/films/2/',
            'https://swapi.dev/api/films/3/',
            'https://swapi.dev/api/films/6/',
          ],
          species: [],
          vehicles: ['https://swapi.dev/api/vehicles/30/'],
          starships: [],
          created: '2014-12-10T15:20:09.791000Z',
          edited: '2014-12-20T21:17:50.315000Z',
          url: 'https://swapi.dev/api/people/5/',
        }}
      />
    )

    const name = screen.getByTestId('name').textContent
    expect(name).toEqual('jendi asa')
  })
})
