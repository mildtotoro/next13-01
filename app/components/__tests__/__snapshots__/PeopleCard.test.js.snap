// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeopleCard renders a card to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="border border-solid border-gray-300 w-[250px] m-3 p-4"
  >
    <h2
      data-testid="name"
    >
      hello
    </h2>
    <h3>
      Birth Year: 19BBY
    </h3>
    <a
      href="https://swapi.dev/api/planets/2/"
      target="_blank"
    >
      Home world
    </a>
    <div />
    <a
      href="https://swapi.dev/api/people/5/"
      target="_blank"
    >
      Read more
    </a>
  </div>
</DocumentFragment>
`;
